### ISSUE TITLE
simulate native <select> dropdown

### ISSUE THREAD

**Issue by eoghanmurray** [rrweb-io/rrweb#457]
A user clicking on a <select> dropdown to look through the options is a significant event (think ecommerce site).
At the moment, I believe this action is not replayable whenever it happens on a native <select> element.
Would it be worthwhile to simulate the appearance of the dropdown in the replayer, even if we likely can't 100% recreate how it looked in the original browser?
We have the list of <options> so could use very generic styling and positioning to show a 'good enough' menu.

I guess we'd have to trigger on a mouseDown event on the <select> and remove the menu after any subsequent mouseUp event (at least that's how it seems to play out in Chromium).

**Comment by startupgurukul** [comment]
I think simulating mouse clicks will take this down into a huge rabbit hole, if shadow-dom is handled will the issues will be resolved?

**Comment by eoghanmurray** [comment]
We are already simulating changing CSS state upon mouse hover, so I'd consider this proposal to be in the same vein.
Actually I don't think I understand your comment at all!  In particular: what is the relevance of shadow-dom?  And what do you mean by read-only?

**Comment by startupgurukul** [comment]
After mouse clicks, there will be changes at the html level ( at a dom level or via shadow dom) is it possible those can be captured so that hopefully there will be no need to simulate mouse clicks and all & just capture the state of webpage.

**Comment by eoghanmurray** [comment]
Ah okay; ya I can't see any DOM changes when a <select> menu is clicked.
Are you saying it's possible to track shadow-dom changes?  
I guess it's related to this one: #38 — you are not working on that?

**Comment by startupgurukul** [comment]
Yes that's the relavent issue, from comments on the thread, this seems helpful https://github.com/medialize/ally.js/blob/master/src/observe/shadow-mutations.js

**Comment by eoghanmurray** [comment]
How would you replay the changes to the shadow dom though?

**Comment by startupgurukul** [comment]
It's almost like normal dom

let shadow = elementRef.attachShadow({mode: 'open'});


While To access the shadowroot 
let myShadowDom = myCustomElem.shadowRoot;

**Comment by startupgurukul** [comment]
Please see my email to hey@getthere.ie Shared an alternative that handles shadow dom. 

I was not sure if it would be appropriate to share an alternative to rrweb in the issues here, undermining the huge effort rrweb authors put in so sent it to you on email directly. I will delete this message right away after you get a notification.

**Comment by startupgurukul** [comment]
Since iframe recording is now possible thanks to the recent commit, hope shadow dom also can also be pulled off.

**Comment by Juice10** [comment]
Looks like we can't record this with Shadow Dom.
Otherwise we could do
`document.querySelector('iframe').contentDocument.querySelector('select').attachShadow({mode: 'open'})` on https://interactive-examples.mdn.mozilla.net/pages/tabbed/select.html
<img width="810" alt="image" src="https://github.com/rrweb-io/rrweb/assets/4106/f916d602-029f-47b8-8bec-1dbbd6d77c79">
(safari)

But maybe we can create our own representation of the dropdown instead

**Comment by Juice10** [comment]
I've also tried simulating mouse clicks in the past, it works for some browsers but doesn't for others.
